src/App.tsx

```diff
@@ -15,48 +16,103 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  empty = '',
+  alpha = 'alpha',
+  length = 'length',
+}
+
+interface SortVer {
+  sortField: SortType,
+  reverseField: boolean,
+}
+
+function gerPreparedGoods(
+  goods: string[],
+  { sortField, reverseField }: SortVer,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods
+      .sort((good1, good2) => {
+        switch (sortField) {
+          case SortType.alpha:
+            return good1.localeCompare(good2);
+
+          case SortType.length:
+            return good1.length - good2.length;
+
+          default:
+            return 0;
+        }
+      });
+  }
+
+  if (reverseField) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState(SortType.empty);
+  const [reverseField, setReverseField] = useState(false);
+
+  const visibleGoods = gerPreparedGoods(
+    goodsFromServer, { sortField, reverseField },
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
+          onClick={() => setSortField(SortType.alpha)}
           type="button"
-          className="button is-info is-light"
+          className={cn('button', 'is-info', {
+            'is-light': sortField !== SortType.alpha,
+          })}
         >
           Sort alphabetically
         </button>
 
         <button
+          onClick={() => setSortField(SortType.length)}
           type="button"
-          className="button is-success is-light"
+          className={cn('button', 'is-success', {
+            'is-light': sortField !== SortType.length,
+          })}
         >
           Sort by length
         </button>
 
         <button
+          onClick={() => (setReverseField(!reverseField))}
```

Comment:
```suggestion
          onClick={() => (setReverseField(true))}
```
There is no toggle button, so you can set value directly